### PR TITLE
Support depositing to Uniswap LPs

### PR DESCRIPTION
**Pull request by jinchung**
### Changes
- Hook up Deposit button in the Pools expanded state
- Include more Exchange Modal types and move over naming of "Deposit" to be more specific to "Deposit Compound" or "Deposit Uniswap" etc

### How it Works
We are using Zapper smart contracts in order to support depositing from a single token into an LP pool (where the Zapper contract manages the internal swaps needed to do so without requiring us to do multiple transactions.) 
There is a "first swap" and a "second swap".
The first swap is from the user selected input currency to one of the pool pair tokens. This trade is not necessarily done on Uniswap, but finds a best quote from 0x. It potentially has the ability to give Rainbow a small trading fee if executed on any of the platforms that provide affiliate fees.
Then, the secondary swap is from half of the output of the first step into the other pair token. This swap is always done using Uniswap.

### Testing
- Depositing to a pool with an input currency that is an ERC20 and is one of the pool assets (i.e, DAI input currency for an ETH-DAI pool)- testing this with and without unlocking required
- Depositing to a pool with an input currency that is an ERC20 but NOT one of the pool assets (i.e, DAI input currency for a ZRX-ETH pool) - testing this with and without unlocking required
- Depositing to a pool with ETH that is one of the pool assets (i.e, ETH input currency for an ETH-DAI pool)
- Depositing to a pool with ETH and is NOT one of the pool assets (i.e, ETH input currency for a USDC-DAI pool)

### CODE REVIEW

**Review by gigilove2001 — APPROVED** [review]

**Review by gigilove2001 — APPROVED** [review]

**Review by gigilove2001 — APPROVED** [review]

File: src/handlers/uniswapLiquidity.ts [hunk 1]
```diff
@@ -1,301 +1,140 @@
...
-      return null;
-    }
-  });
-
-  const results = await Promise.all(promises);
-  const updatedResults = compact(results);
-  return keyBy(updatedResults, result => result.address);
+  if (!firstSwapDetails) return null;
+
+  const minPoolTokens = '1';
```
Comment (jinchung) [inline src/handlers/uniswapLiquidity.ts:56]: Note: This was copying the behavior of Zapper but we should calculate actual min pool token

File: src/components/sheet/sheet-action-buttons/DepositActionButton.js [hunk 2]
```diff
@@ -1,39 +1,46 @@
...
   const color = givenColor || (isDarkMode ? colors.darkModeDark : colors.dark);
-  const [didTrack, setDidTrack] = useState(false);
 
-  const handlePress = useCallback(() => {
-    if (!didTrack) {
-      analytics.track('Tapped placeholder Deposit button', {
-        category: 'liquidity pool',
-        label: symbol,
+  const onDeposit = useCallback(() => {
+    if (!isReadOnlyWallet || enableActionsOnReadOnlyWallet) {
```
Comment (brunobar79) [inline src/components/sheet/sheet-action-buttons/DepositActionButton.js:18]: good call with the isReadOnlyWallet check

File: src/handlers/uniswapLiquidityInfo.ts [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+}
+
+const getTokenDetails = async (
+  chainId: ChainId,
+  tokenAddress: string,
+  pairs: Record<string, Asset>
+): Promise<Asset> => {
+  if (toLower(tokenAddress) === toLower(WETH[chainId].address)) {
+    return {
+      address: 'eth',
```
Comment (brunobar79) [inline src/handlers/uniswapLiquidityInfo.ts:36]: ETH_ADDRESS?

File: src/handlers/uniswapLiquidityInfo.ts [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+      decimals: 18,
+      name: 'Ethereum',
+      symbol: 'ETH',
+    };
+  }
+
+  const tokenContract = new Contract(tokenAddress, erc20ABI, web3Provider);
+
+  const token = get(pairs, `[${toLower(tokenAddress)}]`);
```
Comment (brunobar79) [inline src/handlers/uniswapLiquidityInfo.ts:45]: shall we get rid of all the `get` in this file?

File: src/handlers/uniswapLiquidityInfo.ts [hunk 5]
```diff
@@ -0,0 +1,301 @@
...
+  liquidityPoolTokens: ParsedAddressAsset[],
+  pairs: Record<string, Asset>
+): Promise<Record<string, LiquidityInfo>> => {
+  const liquidityPoolTokensWithNative = parseAssetsNative(
+    liquidityPoolTokens,
+    nativeCurrency
+  );
+  const [v1Tokens, v2Tokens] = partition(
+    liquidityPoolTokensWithNative,
+    token => token.type === 'uniswap'
```
Comment (brunobar79) [inline src/handlers/uniswapLiquidityInfo.ts:120]: Should it be `AssetType.uniswap` ?

File: src/handlers/uniswapLiquidityInfo.ts [hunk 6]
```diff
@@ -0,0 +1,301 @@
...
+
+      return {
+        ...lpToken,
+        tokens: [
+          {
+            balance: tokenBalance,
+            ...token,
+          },
+          {
+            address: 'eth',
```
Comment (brunobar79) [inline src/handlers/uniswapLiquidityInfo.ts:283]: ETH_ADDRESS

File: src/raps/unlockAndDepositUniswap.ts [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+    accountAddress: string;
+  } = store.getState().settings;
+
+  let actions: RapAction[] = [];
+
+  const depositAssetNeedsUnlocking = await assetNeedsUnlocking(
+    accountAddress,
+    inputAmount,
+    inputCurrency,
+    ZapInAddress
```
Comment (brunobar79) [inline src/raps/unlockAndDepositUniswap.ts:75]: I think we have all the const that are addresses using all uppercase chars, which is something that I personally in favor of.

**Review by brunobar79 — APPROVED** [review]
🔥

File: src/handlers/uniswapLiquidityInfo.ts (hunk 3 above)
Comment (jinchung) [reply src/handlers/uniswapLiquidityInfo.ts:36]: ooooh very nice :)

**Pull request closed without merging**